app/Models/Thread.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+
+class Thread extends Model
+{
+    use HasFactory;
```

Comment:
You can use `Diractable` trait to auto generate unique paths to threads. So, we can use `$thread->path` later.